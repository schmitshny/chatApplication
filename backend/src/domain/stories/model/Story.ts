import {
  Model,
  Column,
  Table,
  DataType,
  ForeignKey,
  BelongsTo,
  CreatedAt,
} from 'sequelize-typescript';

import { User } from '../../../domain/user/model/User';

@Table
export class Story extends Model {
  @Column({
    type: DataType.INTEGER,
    allowNull: false,
    primaryKey: true,
    autoIncrement: true,
  })
  id: number;

  @ForeignKey(() => User)
  @Column({
    type: DataType.INTEGER,
    allowNull: false,
  })
  userId: number;

  @BelongsTo(() => User)
  user: User;

  @Column({
    type: DataType.STRING,
    allowNull: false,
  })
  imageUrl: string;

  @CreatedAt
  createdAt: Date;
}
