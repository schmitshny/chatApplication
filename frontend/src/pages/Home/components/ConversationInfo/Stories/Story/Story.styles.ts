import styled from 'styled-components';

export const StyledStory = styled.div`
  border-radius: 50%;
  border: 2px solid;
  border-color: linear-gradient(to right, red, orange, yellow, green, blue, indigo, violet) 1;
  display: flex;
  align-items: center;
  justify-content: center;
  position: relative;
  cursor: pointer;
`;
