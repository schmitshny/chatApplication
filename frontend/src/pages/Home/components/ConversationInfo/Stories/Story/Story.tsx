import { Avatar } from '../../../../../../components';
import { StyledStory } from './Story.styles';

interface StoryProps {
  storyUrl: string;
  onClick?: () => void;
}

export const Story = ({ storyUrl, onClick }: StoryProps) => {
  return (
    <StyledStory onClick={onClick} data-testid="single-story">
      <Avatar size="medium" shape="circle" avatarUrl={storyUrl} />
    </StyledStory>
  );
};
