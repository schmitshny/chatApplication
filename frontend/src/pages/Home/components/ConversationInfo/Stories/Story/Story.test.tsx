import { describe, test, expect, vi } from 'vitest';
import { fireEvent, render, screen } from '@testing-library/react';
import { Story } from './Story';

describe('Story', () => {
  test('renders with provided story URL', () => {
    const testUrl = 'http://example.com/story1.jpg';
    const { container } = render(<Story storyUrl={testUrl} />);

    const storyElement = screen.getByTestId('single-story');
    expect(storyElement).toBeInTheDocument();
    expect(storyElement.querySelector('img')).toHaveAttribute('src', testUrl);
    expect(container).toMatchSnapshot();
  });

  test('calls onClick when story is clicked', () => {
    const handleClick = vi.fn();
    render(<Story storyUrl="http://example.com/story.jpg" onClick={handleClick} />);

    const storyElement = screen.getByTestId('single-story');
    fireEvent.click(storyElement);

    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
