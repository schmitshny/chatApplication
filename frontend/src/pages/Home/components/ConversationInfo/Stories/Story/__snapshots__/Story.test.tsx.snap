// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Story > renders with provided story URL 1`] = `
<div>
  <div
    class="sc-bZkfhe jitLc"
    data-testid="single-story"
  >
    <div
      class="sc-jSMdHm eJItaO"
    >
      <img
        alt="user avatar"
        class="sc-gKXNOb gUEVCF"
        loading="lazy"
        shape="circle"
        src="http://example.com/story1.jpg"
      />
    </div>
  </div>
</div>
`;
